lnst/Controller/Recipe.py

```diff
@@ -216,7 +216,17 @@ def results(self):
 
     @property
     def overall_result(self):
-        return all([i.success for i in self.results] + [self.exception is None])
+        results = [i.result for i in self.results]
+
+        if (ResultType.FAIL in results) or (self.exception is not None):
+            return ResultType.FAIL
+
+        if ResultType.WARNING in results:
+            return ResultType.WARNING
+        elif all([i == ResultType.PASS for i in results]):
+            return ResultType.PASS
```

Comment:
i think this can all be simplified by using the `max_severity` class method and https://docs.python.org/3/library/functools.html?highlight=reduce#functools.reduce

Reply:
so i've replaced property `succes` with `result` as it could also return `WARNING` not just pass/fail